driver/terraform.go

```diff
@@ -65,71 +65,52 @@ type TerraformConfig struct {
-	return &Terraform{
-		log:               config.Log,
-		persistLog:        config.PersistLog,
-		path:              config.Path,
-		workingDir:        config.WorkingDir,
-		backend:           config.Backend,
-		requiredProviders: config.RequiredProviders,
-		clientType:        config.ClientType,
-	}
-}
-
-// Init initializes the Terraform local environment. The Terraform binary is
-// installed to the configured path.
-func (tf *Terraform) Init(ctx context.Context) error {
-	if _, err := os.Stat(tf.workingDir); os.IsNotExist(err) {
-		if err := os.MkdirAll(tf.workingDir, workingDirPerms); err != nil {
-			log.Printf("[ERR] (driver.terraform) error creating base work directory: %s", err)
-			return err
+func NewTerraform(config *TerraformConfig) (*Terraform, error) {
+	if _, err := os.Stat(config.WorkingDir); os.IsNotExist(err) {
+		if err := os.Mkdir(config.WorkingDir, workingDirPerms); err != nil {
+			log.Printf("[ERR] (driver.terraform) error creating task work directory: %s", err)
+			return nil, err
 		}
 	}
 
-	if isTFInstalled(tf.path) {
-		tfVersion, compatible, err := isTFCompatible(ctx, tf.workingDir, tf.path)
-		if err != nil {
-			if strings.Contains(err.Error(), "exec format error") {
-				return errIncompatibleTerraformBinary
-			}
-			return err
-		}
-
-		if !compatible {
-			return errUnsupportedTerraformVersion
-		}
-
-		tf.version = tfVersion.String()
-		log.Printf("[INFO] (driver.terraform) skipping install, terraform %s "+
-			"already exists at path %s/terraform", tf.version, tf.path)
-		return nil
+	tfClient, err := newClient(&clientConfig{
+		task:       config.Task,
+		clientType: config.ClientType,
+		log:        config.Log,
+		persistLog: config.PersistLog,
+		path:       config.Path,
+		workingDir: config.WorkingDir,
+	})
+	if err != nil {
+		log.Printf("[ERR] (driver.terraform) init client type '%s' error: %s", config.ClientType, err)
+		return nil, err
 	}
 
-	log.Printf("[INFO] (driver.terraform) installing terraform to path '%s'", tf.path)
-	if err := tf.install(ctx); err != nil {
-		log.Printf("[ERR] (driver.terraform) error installing terraform: %s", err)
-		return err
+	handler, err := getTerraformHandlers(config.Task)
+	if err != nil {
+		return nil, err
 	}
-	log.Printf("[INFO] (driver.terraform) successfully installed terraform")
-	return nil
+
+	return &Terraform{
+		task:              config.Task,
+		backend:           config.Backend,
+		requiredProviders: config.RequiredProviders,
+		workingDir:        config.WorkingDir,
+		worker:            newWorker(defaultRetry),
+		client:            tfClient,
+		postApply:         handler,
+	}, nil
 }
 
 // Version returns the Terraform CLI version for the Terraform driver.
 func (tf *Terraform) Version() string {
-	return tf.version
+	return TerraformVersion
 }
 
 // InitTask initializes the task by creating the Terraform root module, creating
 // client to execute task, and retrieving post-Terraform apply handlers
```

Comment:
I think you could remove the stuff about creating client and retrieving the handlers from the comment

Reply:
Appreciate your attention to detail, i missed these comments when updating. Will do!